File: chapter 2/learning_topic/scripts/turtle_publisher.py
#!/usr/bin/env python
# coding:utf-8

import rospy
from geometry_msgs.msg import Twist

def velocity_publisher():
	
    rospy.init_node('turtle1_vel_publisher', anonymous=True)

    turtle1_vel_pub = rospy.Publisher('/turtle1/cmd_vel', Twist, queue_size=1000)

    rate = rospy.Rate(10) 
    
    count = 0
    while not rospy.is_shutdown():

        vel_msg = Twist()
        vel_msg.linear.x = 1.0
        vel_msg.angular.z = 1.0

        turtle1_vel_pub.publish(vel_msg)
        rospy.loginfo("Publish turtle1 velocity msssage is：[%0.2f m/s, %0.2f rad/s]", vel_msg.linear.x, vel_msg.angular.z)  
        count+=1
        rospy.loginfo("Publish times are:[%d]", count)
        rate.sleep()

if __name__ == '__main__':
    try:
        velocity_publisher()
    except rospy.ROSInterruptException:
        pass

File: chapter 2/learning_topic/src/turtle_publisher.cpp
#include <ros/ros.h> 
#include <geometry_msgs/Twist.h>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "turtle1_vel_publisher");

	ros::NodeHandle nh;

	ros::Publisher turtle1_vel_pub = nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 1000);

	ros::Rate loop_rate(10);

	int count = 0;
	while (ros::ok())
	{
		geometry_msgs::Twist vel_msg;
		vel_msg.linear.x = 1.0;
		vel_msg.angular.z = 1.0;

		turtle1_vel_pub.publish(vel_msg);
		ROS_INFO("Publish turtle1 velocity msssage is：[%0.2f m/s, %0.2f rad/s]", vel_msg.linear.x, vel_msg.angular.z);
        ROS_INFO("Publish times are:[%d]",++count);
        
	    loop_rate.sleep();
	}

	return 0;
}